tests/Backend/Snowflake/WhereFilterTest.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\Snowflake;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\TableExporter;
+use Keboola\Test\StorageApiTestCase;
+
+class WhereFilterTest extends StorageApiTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+        $this->_initEmptyTestBuckets();
+    }
+
+    public function testSimpleWhereConditions()
+    {
+        $tableId = $this->prepareTable();
+
+        $where = [
+            [
+                "column" => "column_string",
+                "operator" => "eq",
+                "values" => ["first"],
+            ],
+        ];
+        $preview = $this->_client->getTableDataPreview($tableId, ['whereFilters' => $where]);
+        $this->assertCount(1, Client::parseCsv($preview));
+        $this->assertCount(1, $this->getExportedTable($tableId, ['whereFilters' => $where]));
+    }
+
+    public function testFilterWithCast()
+    {
+        $tableId = $this->prepareTable();
+
+        $where = [
+            [
+                'column' => 'column_string_number',
+                'operator' => 'ge',
```

Comment:
Mohl bych tady udělat testy na všechny OP, ale myslím, že to stačí pokrytý unit testama v KBC

Reply:
cajk